File: frontend_app/job_portal/lib/models/merged_interaction.dart
import 'package:job_portal/models/applicant_model.dart';
import 'package:job_portal/models/job_model.dart';

class MergedInteraction {
  int? id;
  Applicant? applicantId;
  Job? jobId;
  String? statusApply;
  int? isSave;
  int? isRead;

  MergedInteraction(
      {this.id,
      this.applicantId,
      this.jobId,
      this.statusApply,
      this.isSave,
      this.isRead});

  MergedInteraction.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    applicantId = json['applicantId'] != null
        ? new Applicant.fromJson(json['applicantId'])
        : null;
    jobId = json['jobId'] != null ? new Job.fromJson(json['jobId']) : null;
    statusApply = json['statusApply'];
    isSave = json['isSave'];
    isRead = json['isRead'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    if (this.applicantId != null) {
      data['applicantId'] = this.applicantId!.toJson();
    }
    if (this.jobId != null) {
      data['jobId'] = this.jobId!.toJson();
    }
    data['statusApply'] = this.statusApply;
    data['isSave'] = this.isSave;
    data['isRead'] = this.isRead;
    return data;
  }
}

File: backend/jobporal/src/main/java/com/project/jobportal/models/MergedInteraction.java
package com.project.jobportal.models;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "merged_interactions")
public class MergedInteraction {

    @Id
    @Column(name = "id") // hoặc một cột duy nhất đại diện (bắt buộc cần @Id cho JPA)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "applicant_id")
    private Applicants applicantId;

    @ManyToOne
    @JoinColumn(name = "job_id")
    private Jobs jobId;

    @Column(name = "status_apply")
    private String statusApply;//Pending, Approved, Rejected

    @Column(name = "is_save")
    private Long isSave;

    @Column(name = "is_read")
    private Long isRead;

    @Column(name = "is_lock")
    private Long isLock;

    @Column(name = "status")
    private String status;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
